aspnetcore/performance/performance-best-practices/samples/3.0/Controllers/MyFirstController.cs

```diff
@@ -27,11 +27,25 @@ public class GoodStreamReaderController : Controller
         [HttpGet("/contoso")]
         public async Task<ActionResult<ContosoData>> Get()
         {
-            return await JsonSerializer.DeserializeAsync<ContosoData>(Request.Body);
+            var json = await new StreamReader(Request.Body).ReadToEndAsync();
+            
+            return await System.Text.Json.JsonSerializer.Deserialize<ContosoData>(json);
         }
 
     }
     #endregion
+    
+    #region snippet3
+    public class GoodStreamReaderController : Controller
+    {
+        [HttpGet("/contoso")]
+        public async Task<ActionResult<ContosoData>> Get()
+        {
+            return await System.Text.Json.JsonSerializer.DeserializeAsync<ContosoData>(Request.Body);
```

Comment:
Same as above. Add a using.

Reply:
Just updated

Reply:
`snippet3` needs to move to a different namespace so it compiles. It's not clear why you're adding `snippet3`